PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/console/SparkScalaLivyConsoleRunConfiguration.kt

```diff
@@ -83,24 +96,78 @@ open class SparkScalaLivyConsoleRunConfiguration(project: Project,
+            }
+
+            return SparkScalaLivyConsoleRunProfileState(consoleBuilder, sparkSession)
+        } catch (err: Throwable) {
+            throw ExecutionException(err)
+        }
+    }
+
+    open fun createSession(sparkCluster: IClusterDetail): SparkSession {
+        val url = URI.create((sparkCluster as? LivyCluster)?.livyConnectionUrl
+                ?: throw RuntimeConfigurationError("Can't prepare Spark interactive session since Livy URL is empty"))
+
+        return if (sparkCluster is MfaEspCluster) MfaEspSparkSession(
                 name,
                 url,
-                cluster.tenantId
-        ) else SparkSession(
+                sparkCluster.tenantId)
+        else SparkSession(
                 name,
                 url,
-                cluster.httpUserName,
-                cluster.httpPassword)
+                sparkCluster.httpUserName,
+                sparkCluster.httpPassword)
+    }
+
+    private val batchSubmitModel: SparkSubmitModel
+            get() = this.batchRunConfiguration?.submitModel
+                    ?: throw RuntimeConfigurationError("Can't find Spark batch job configuration to inherit")
 
-        return SparkScalaLivyConsoleRunProfileState(consoleBuilder, session)
+    open fun applyRunConfiguration(sparkCluster: IClusterDetail, session: SparkSession, deploy: Deployable) {
+        session.jars = batchSubmitModel.referenceJars
+        session.files = batchSubmitModel.referenceFiles
+        session.conf = batchSubmitModel.jobConfigs.map { it[0] to it[1] } .toMap()
+
+        batchSubmitModel.artifactPath.ifPresent {
+            session.deploy = deploy
+            synchronized(session.artifactsToDeploy) {
+                session.artifactsToDeploy.clear()
+                session.artifactsToDeploy.add(it)
+            }
+        }
+
+        if (!batchSubmitModel.isLocalArtifact) {
+            ArtifactUtil.getArtifactWithOutputPaths(project)
+                    .first { artifact -> artifact.name == batchSubmitModel.artifactName }
+                    ?.let { setBuildArtifactBeforeRun(project, this, it) }
+        }
     }
 
-    override fun checkRunnerSettings(runner: ProgramRunner<*>, runnerSettings: RunnerSettings?, configurationPerRunnerSettings: ConfigurationPerRunnerSettings?) {
-        cluster = ClusterManagerEx.getInstance().getClusterDetailByName(clusterName)
-                .orElseThrow { RuntimeConfigurationError("Can't find the target cluster $clusterName") }
+    override fun checkRunnerSettings(runner: ProgramRunner<*>,
+                                     runnerSettings: RunnerSettings?,
+                                     configurationPerRunnerSettings: ConfigurationPerRunnerSettings?) {
+        batchRunConfiguration?.checkRunnerSettings(runner, runnerSettings, configurationPerRunnerSettings)
+    }
+
+    override fun prepare(runner: ProgramRunner<RunnerSettings>): Observable<SparkScalaLivyConsoleRunConfiguration> {
```

Comment:
Just in curious, where is this `prepare` method get called?

Reply:
Aligning with Spark batch job, invoked by `RunConfigurationActionUtils::checkAndPrepareRunProfileState()` in `,runEnvironmentProfileWithCheckSettings`